File: firmware/test_oled.py
"""Stage 3 roadmap task 17: test the OLED display independently.

Requires the micropython-ssd1306 library installed on the device
(Thonny -> Tools -> Manage packages -> micropython-ssd1306).
Run on-device and record the observed output as evidence in
docs/PROJECT_STATE.md before marking this task done.
"""
from machine import Pin, I2C
from ssd1306 import SSD1306_I2C

i2c = I2C(0, scl=Pin(9), sda=Pin(8), freq=400000)
oled = SSD1306_I2C(128, 64, i2c)

oled.fill(0)
oled.text("AgriControl", 0, 0, 1)
oled.text("OLED test", 0, 16, 1)
oled.text("Stage 3 task 17", 0, 32, 1)
oled.show()

File: firmware/src/test_oled.cpp
// Stage 3 roadmap task 17: test the OLED display independently (env:test_oled).
//
// Build/upload with: pio run -e test_oled -t upload
// Record the observed output as evidence in docs/PROJECT_STATE.md before
// marking this task done.
#include <Arduino.h>
#include <U8g2lib.h>
#include <Wire.h>

#include "pins.h"

U8G2_SSD1306_128X64_NONAME_F_HW_I2C u8g2(U8G2_R0, U8X8_PIN_NONE);

void setup() {
  Wire.begin(pins::I2C_SDA, pins::I2C_SCL);
  u8g2.begin();
  u8g2.clearBuffer();
  u8g2.setFont(u8g2_font_6x10_tf);
  u8g2.drawStr(0, 12, "AgriControl");
  u8g2.drawStr(0, 26, "OLED test");
  u8g2.drawStr(0, 40, "Stage 3 task 17");
  u8g2.sendBuffer();
}

void loop() {}
